tests/generation/test_generation_utils.py

```diff
@@ -697,31 +697,43 @@ def test_sample_generate(self):
             logits_warper_kwargs, logits_warper = self._get_warper_and_kwargs(num_beams=1)
 
             # check `generate()` and `sample()` are equal
-            output_sample, output_generate = self._sample_generate(
-                model=model,
-                input_ids=input_ids,
-                attention_mask=attention_mask,
-                max_length=max_length,
-                num_return_sequences=1,
-                logits_processor=logits_processor,
-                logits_warper=logits_warper,
-                logits_warper_kwargs=logits_warper_kwargs,
-                process_kwargs=process_kwargs,
-            )
+            try:
```

Comment:
Hmm, this is a bit dangerous IMO. If generate fails because of `-inf` values - it can also very well indicate that there is a bug in the code or some other kind of problem. Could we maybe instead try to create dummy models that have a much lower chance of creating -inf values? 

Happy to help on this front. It should be mostly Pegasus (and any other models?)

Reply:
Hey @patrickvonplaten.

The test method `test_sample_generate`  prepares random inputs (`self.model_tester.prepare_config_and_inputs_for_common()`). I am afraid the attempt to `lower chance of creating -inf values` won't be very realistic.

One thing that might work is to set a seed in `test_sample_generate`, before calling `_get_input_ids_and_config`, in order to get a fixed input ids that we know it will work.

Do you have maybe other ideas?

FYI: The current script to create dummy models is a bit messy so far. I have some work on it a few months ago, but not merged to main yet.

Reply:
Can we pinpoint which models tend to fail here (I also only seem to see the Pegasus model to fail here - are there other ones?)

Let's maybe try to fix them one by one by changing the model config

Reply:
From Slack reports, I see `Marian`, `MBart`, `Blenderbot` and `BlenderbotSmall`. I didn't see `Pegasus`, but Matthijs mentioned it in one message.

Reply:
Thought about it a bit more and might have a hunch of what's going on here. Could we try to solve the problem via this PR: https://github.com/huggingface/transformers/pull/18053 that disables force generating tokens and see whether this might solve the problem? Given that it's mostly Bart-like models that show flaky behavior is an indication that it's related to the `forced_eos_token_id` feature of BART